base/poco/Util/src/AbstractConfiguration.cpp

```diff
@@ -529,4 +564,122 @@ void AbstractConfiguration::setRawWithEvent(const std::string& key, std::string
 }
 
 
+void AbstractConfiguration::checkHostValidity(const std::string& value)
+{
+	if (!isValidIPv4Address(value) && !isValidIPv6Address(value) && !isValidDomainName(value))
+	{
+		throw SyntaxException("Property is not a valid host name", value);
+	}
+}
+
+
+bool AbstractConfiguration::isValidIPv4Address(const std::string& value)
+{
+	int octet = 0;
```

Comment:
Can we use https://github.com/ClickHouse/ClickHouse/blob/ca514c655687946a9d762252b1e30049c560981e/base/poco/Net/src/IPAddressImpl.cpp#L256 and https://github.com/ClickHouse/ClickHouse/blob/ca514c655687946a9d762252b1e30049c560981e/base/poco/Net/src/IPAddressImpl.cpp#L609?

Reply:
Yes, I think it's a good idea. Thank you!